modules/router/src/services/autoRebalance.ts

```diff
@@ -544,14 +668,37 @@ const sendTransaction = async (
     },
     "Tx sent",
   );
-  const receipt = await response.wait(getConfirmationsForChain(chainId));
+  return response.hash;
+};
+
+
+const getConfirmation = async (
+  txHash: string,
+  swap: AllowedSwap,
+  providers: HydratedProviders,
+  logger: BaseLogger,
+  method: string = "getConfirmation",
+  methodId: string = getRandomBytes32()
+): Promise<Result<TransactionReceipt, AutoRebalanceServiceError>> => {
+  const result = await waitForTransaction(providers[swap.fromChainId], txHash, getConfirmationsForChain(swap.fromChainId));
```

Comment:
Good spot, thanks. Yeah I really wasn't sure for this. How can we determine which chain is the right one here?

Keep in mind we may be 'coming back' at a later point to `waitForTransaction` - so it's not necessarily like I can just re-use the TransactionReceipt object (from ethersproject)

Reply:
Hmmm. I wonder if you can use a transaction hash to be a string that has `hash,chainId` or just add it to the db model since now this will need to be stateful.

I guess if we want to just code in our assumptions it would be:
1. `approve` on `from` chain
2. `execute` on `from` chain
3. `complete` on `to` chain

Reply:
Current solution implemented:
- added `approveChain`, `executeChain`, `completeChain` to the schema
- filled in those values whenever respective hash was saved.
- used those values for confirmation, guaranteeing ability to confirm separately and safely across method calls.